src/main/webapp/app/entities/programming-exercise/actions/programming-exercise-trigger-build-button.component.ts

```diff
@@ -82,4 +84,28 @@ export abstract class ProgrammingExerciseTriggerBuildButtonComponent implements
             )
             .subscribe();
     }
+
+    /**
+     * Trigger a regular build or a failed build, depending on the state of the latest submission.
+     *
+     * @param submissionType that is used for the creation of the submission.
+     */
+    triggerBuild(submissionType: SubmissionType) {
+        this.isRetrievingBuildStatus = true;
+        if (this.participationHasLatestSubmissionWithoutResult) {
+            this.submissionService
+                .triggerFailedBuild(this.participation.id)
+                .pipe(
+                    tap(res => {
+                        // This status code means that a build is already queued/running for the given participation.
+                        if (res.status === 202) {
+                            this.isBuilding = true;
```

Comment:
I guess it would be good to reset/start the timer in `submissionService` when this happens so that the `Building and testing ... " animation appears again.
Or is this done automatically when setting `isBuilding` to true? (then documentation would be good)

Reply:
I looked at the implementation and improved it: I now send the submission again to the client which resets the timer in the programming-submission.service. https://github.com/ls1intum/Artemis/pull/996/commits/8a5a5d09164b3fa82b12503ce2264d692235d353

Therefore the isBuilding flag will automatically be set as the programming-submission.service will emit the submission again to the subscribers.